typescript/cli/src/submit/submit.ts

```diff
@@ -40,32 +40,37 @@ export async function getSubmitterBuilder<TProtocol extends ProtocolType>({
 async function getSubmitter<TProtocol extends ProtocolType>(
   multiProvider: MultiProvider,
   submitterMetadata: SubmitterMetadata,
+  isDryRun = false,
 ): Promise<TxSubmitterInterface<TProtocol>> {
   switch (submitterMetadata.type) {
     case TxSubmitterType.JSON_RPC:
       return new EV5JsonRpcTxSubmitter(multiProvider);
     case TxSubmitterType.IMPERSONATED_ACCOUNT:
+      if (!isDryRun)
+        throw new Error(
+          'Impersonated account submitters may only be used during dry-runs.',
+        );
       return new EV5ImpersonatedAccountTxSubmitter(
         multiProvider,
-        submitterMetadata.props as EV5ImpersonatedAccountTxSubmitterProps,
+        submitterMetadata.props,
       );
     case TxSubmitterType.GNOSIS_SAFE:
       return new EV5GnosisSafeTxSubmitter(
         multiProvider,
-        submitterMetadata.props as EV5GnosisSafeTxSubmitterProps,
+        submitterMetadata.props,
       );
     default:
-      throw new Error(`Invalid TxSubmitterType: ${submitterMetadata.type}`);
+      throw new Error(`Invalid TxSubmitterType.`);
```

Comment:
Why remove the extra info here?

Reply:
thanks to zod, it's impossible to have the `type` be set incorrectly (you'll find the compiler now catches this). yes this implies we could remove the `default` case altogether, but being explicit about what this default case implies is a tradeoff I chose to make here (even though it will never be reached)